src/services/textChanges.ts

```diff
@@ -64,6 +64,13 @@ namespace ts.textChanges {
         return skipTrivia(text, start, /*stopAfterLineBreak*/ false, /*stopAtComments*/ true);
     }
 
+    function rangeOfTypeParameters(text: string, typeParameters: NodeArray<TypeParameterDeclaration>): TextRange {

```

Comment:
I think you should make this fix in utilities.ts instead of making a copy and fixing it.

Reply:
The `checker.ts` uses `rangeOfTypeParameters()` from `compiller/utilities` without `sourceFile`, which is required in service.